nflow-rest-api-spring-web/src/main/java/io/nflow/rest/v1/springweb/SpringWebResource.java

```diff
@@ -0,0 +1,34 @@
+package io.nflow.rest.v1.springweb;
+
+import static org.springframework.http.HttpStatus.BAD_REQUEST;
+import static org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR;
+import static org.springframework.http.HttpStatus.NOT_FOUND;
+import static org.springframework.http.ResponseEntity.status;
+
+import java.util.function.Supplier;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+import io.nflow.engine.service.NflowNotFoundException;
+import io.nflow.rest.v1.ResourceBase;
+import io.nflow.rest.v1.msg.ErrorResponse;
+
+public abstract class SpringWebResource extends ResourceBase {
+
+  protected ResponseEntity<?> handleExceptions(Supplier<ResponseEntity<?>> response) {
+    try {
+      return response.get();
+    } catch (IllegalArgumentException e) {
+      return toErrorResponse(BAD_REQUEST, e.getMessage());
+    } catch (NflowNotFoundException e) {
+      return toErrorResponse(NOT_FOUND, e.getMessage());
+    } catch (Throwable t) {
+      return toErrorResponse(INTERNAL_SERVER_ERROR, t.getMessage());
+    }
+  }
+
+  private ResponseEntity<?> toErrorResponse(HttpStatus status, String error) {
```

Comment:
Only this method is different between sprinweb and jaxrs resource. Maybe move the handleException to ResourceBase and make toErrorResponse abstract there?

Reply:
You would need to make ResourceBase generic so that you pass the Response object class there type parameter.

Reply:
I was thinking about that, but also the statuses are different (HttpStatus vs Response.Status). But if I use ints instead of those, maybe it works. I'll give it a try.